src/onegov/form/display.py

```diff
@@ -121,6 +122,78 @@ def __call__(self, field: 'Field') -> Markup:
         ).format(url=field.data)
 
 
+@registry.register_for('VideoURLField')
+class VideoURLFieldRenderer(BaseRenderer):
+    """
+    Renders a video url. Embeds the video if in case of vimeo or
+    youtube otherwise just displays the url as a link.
+    """
+
+    video_template = """
+        <div class="video">
+            <h2 i18n:translate="" class="visually-hidden">Video</h2>
+            <div class="videowrapper">
+                <iframe allow="fullscreen"
+                frameborder="0" src="${url}"></iframe>
+            </div>
+        </div>
+    """
+    url_video_template = """<a href=${url}>${url}</a>"""
+
+    video_page = PageTemplate(video_template)
+    url_video_page = PageTemplate(url_video_template + video_template)
```

Comment:
For such simple templates Chameleon seems like complete overkill, just use `Markup` directly. You will need an additional replacement variable for the translation, but otherwise lingua won't pick up `Video` as a translation string anyways.

```suggestion
    video_template = Markup("""
        <div class="video">
            <h2 class="visually-hidden">{title}</h2>
            <div class="videowrapper">
                <iframe allow="fullscreen"
                frameborder="0" src="{url}"></iframe>
            </div>
        </div>
    """)
    url_video_template = Markup("""<a href="{url}">{url}</a>""")
```
Either way you don't need to manually escape, both Chameleon and Markup escape format strings automatically.

Also it's worth noting that the use of iframe is a bit questionable due to tracking, especially in non-public directories, can be sandbox the iframe for both YouTube and Vimeo to a degree where we don't have to worry about this?

Reply:
Agree, just had mypy complaining about it 'MS001 Markup is used in a dangerous way' :-)

Reply:
That's a flake8 plugin, not mypy. Although mypy could potentially catch this error in the future via `LiteralString`.